### ISSUE TITLE
Zoom in/out action of compass on map broken by ace_map

### ISSUE THREAD

**Issue by severgun** [acemod/ACE3#6200]
**Arma 3 Version:** `1.80` (stable)
**CBA Version:** `3.5` (stable)
**ACE3 Version:** `3.12.1` (stable)

**Mods:**
```
- CBA_A3
- ace
```

**Description:**
- Zoom in/out action of compass on map broken by ace_map. Once you double click and move it, you can't zoom in/out it again, because Arma overwrite position[] and positionBack[] at Arma3Profile on disk.

**Steps to reproduce:**
- Double click on compass, move it around, double click again and again and again.

**Where did the issue occur?**
- Everywhere

**Placed Modules:**
- none

**Solution**
- Seems like setting zoomDuration to **non zero** value will fix this. (I tried 0.1)
https://github.com/acemod/ACE3/blob/2cad71cd1e42338fb757ff1ea22b53d44d401bcc/addons/map/config.cpp#L103

**Comment by commy2** [comment]
Isn't this intentional?

**Comment by severgun** [comment]
What intentional? Line 102 make it bigger. But you can stuck with zoomed out which is not readable. And all your efforts in line 102 went to hell.

So IMHO author want make it bigger and disable animation of resizing. But zero value gives undefined behavior.

**Comment by PabstMirror** [comment]
![compass](https://user-images.githubusercontent.com/9376747/37728321-f3edacb2-2d07-11e8-86e0-9ee297b95c2f.png)

Vanilla Small
Vanilla Big
ACE Normal (it's locked to this size)
ACE - Big (keep scale change, but allow zooming)
```
            scale = 0.7;
            // zoomDuration = 0;
```

So I guess we wanted to remove the compass zooming and just leave it the big size all the time.
ACE does make it as big as you could get from vanilla
There does seem to be some weirdness in the profile, I think people would have to change it to fix zooming.

**Comment by severgun** [comment]
I can't get why not to allow zoom action? Why compass should stay at one size?
Also zoomDuration did not disable zoom action even if it was desired.

Now it can stuck at vanilla small.

**Comment by jonpas** [comment]
Because zooming compass is broken in the game. It can get stuck in small mode.

**Comment by commy2** [comment]
Isn't this also about the ruler being scaled correctly?

**Comment by severgun** [comment]
BIS Fault?
How to reproduce stuck on vanilla?

**Comment by commy2** [comment]
Never heard of that. It doesn't scale so it fits the map grid.

**Comment by barman75** [comment]
Still no solution?
I have to edit `class Compass` in *.Arma3Profile file every time ACE breaks the code, to fix the position and size.

**Comment by commy2** [comment]
It's not an issue, so there can be no solution.

**Comment by severgun** [comment]
> Still no solution?
Solution in ticket. Just create your own pbo which will overwrite ACE "feature".

**Comment by dedmen** [comment]
Arma stores the compass position and size in front and back mode (zoomed in and out)
This ace feature of changing the transition duration to instant (which i wouldn't consider a important feature, 0.1 or 0.3 seconds sounds just fine to me?) breaks this Arma feature and can cause Arma to store the same position for both front and back, meaning you'll forever get stuck in either zoomed, or unzoomed and double clicking doesn't do anything as the other zoom mode has the exact same size stored in Arma profile.

This is an issue for every Control with zoomable object (its a base UI component) so BI cannot add a compass specific workaround for it.

**Comment by jonpas** [comment]
Fixed by #8176.
